src/index.ts

```diff
@@ -12,63 +13,54 @@ import { IKernel, IKernelSpecs } from '@jupyterlite/kernel';
 
 import { WebWorkerKernel } from './web_worker_kernel';
 
-const EXTENSION_NAME = 'xeus';
-const EXTENSION_STATIC_DIR = `../extensions/@jupyterlite/${EXTENSION_NAME}/static/`;
-
 // helper function to fetch json
-function getPkgJson(url: string) {
-  const json_url = EXTENSION_STATIC_DIR + url;
-  const xhr = new XMLHttpRequest();
-  xhr.open('GET', json_url, false);
-  xhr.send(null);
-  return JSON.parse(xhr.responseText);
-}
+async function getJson(url: string) {
+  const json_url = URLExt.join(
+    PageConfig.getBaseUrl(),
+    url,
+  );
 
-let kernel_dir: string[] = [];
-try {
-  kernel_dir = getPkgJson('share/jupyter/kernels.json');
-} catch (err) {
-  console.log(err);
-  console.log('could not fetch share/jupyter/kernels/kernels.json');
-  kernel_dir = [];
-  throw err;
+  const response = await fetch(json_url);
+  return JSON.parse(await response.text());
 }
 
-// fetch kernel spec for each kernel
-const kernel_specs = kernel_dir.map(kernel_dir => {
-  const spec: any = getPkgJson(
-    'share/jupyter/kernels/' + kernel_dir + '/kernel.json'
-  );
-  spec.name = kernel_dir;
-  spec.dir = kernel_dir;
-  spec.resources = {
-    'logo-32x32':
-      EXTENSION_STATIC_DIR +
-      'share/jupyter/kernels/' +
-      kernel_dir +
-      '/logo-32x32.png',
-    'logo-64x64':
-      EXTENSION_STATIC_DIR +
-      'share/jupyter/kernels/' +
-      kernel_dir +
-      '/logo-64x64.png'
-  };
-  return spec;
-});
+export const xeusKernelsPlugin: JupyterLiteServerPlugin<void> = {
```

Comment:
I also reworked the plugins, exposing just one plugin from now on, and making a non-blocking request to the server for `kernels.json`

Reply:
Having separate plugins for each kernels could also be useful, for example if someone would like to disable some kernels for a particular app (for instance `repl`), while keeping all the kernels for the full `lab` app?

Reply:
Right!

Are we able to register plugins asynchronously? I understand we can't, so we may not be able to get rid of the blocking XmlHTTPRequest?

Reply:
Another idea (briefly mentioned in https://github.com/jupyterlite/xeus/pull/4#issuecomment-1851976232) could be to generate that index file at `jupyter lite build` time (via a template).

The CLI addon would then need to place that generated file correctly so it can be picked up as a JupyterLab extension (normally it already knows about the location of labextensions under the prefix).

Reply:
> I'm not a big fan of generating this at build time.

Sure, because of the additional complexity, or other reasons?

Reply:
For the additional complexity, if we really want to be able to disable kernels individually, I guess I prefer the blocking HTTP request over more templating code.

Reply:
Actually there is another issue.

The kernel specs are polled every one minute https://github.com/jupyterlab/jupyterlab/blob/main/packages/services/src/kernelspec/manager.ts#L42 so registering specs asynchronously is NOT the way to go if we don't want to wait for a minute before getting our kernels displayed.

Let's keep the blocking call for now.

Reply:
Thinking more about it, would generating the files at `jupyter lite build` time add that much complexity?

We could imagine having the addon generate one file per available kernel, for example `xeus-python.js`, `xeus-lua.js` and so on. The only difference between these files would be name of the kernel. And having an `index.js` that would expose the plugins, for example:

```js
import xeusPython from 'xeus-python';
import xeusLua from 'xeus-lua';

const plugins = [xeusPython, xeusLua];
export default plugins;
```

Reply:
> Actually there is another issue.
> 
> The kernel specs are polled every one minute https://github.com/jupyterlab/jupyterlab/blob/main/packages/services/src/kernelspec/manager.ts#L42 so registering specs asynchronously is NOT the way to go if we don't want to wait for a minute before getting our kernels displayed.
> 
> Let's keep the blocking call for now.

Just keep in mind how minuscule the files are. There is nooo way that blocking is a problem at all

Reply:
> Thinking more about it, would generating the files at jupyter lite build time add that much complexity?

In the end, because we need to keep the blocking call, I will revert the changes from this files for now and keep the individual plugins per-kernel.